code/modules/projectiles/cell_cartridges/cell_cartridges.dm

```diff
@@ -0,0 +1,140 @@
+//Boxes of ammo
+/obj/item/cell_cartridge
+	name = "cell cartridge"
+	desc = "A cartridge for holding power cells; typically used with energy weapons."
+	icon = 'icons/obj/ammo.dmi'
+	icon_state = "cell_cartridge"
+	flags_1 = CONDUCT_1
+	slot_flags = ITEM_SLOT_BELT
+	item_state = "syringe_kit"
+	lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/medical_righthand.dmi'
+	materials = list(/datum/material/plastic = 5000, /datum/material/glass = 1000, /datum/material/gold = 1000)
+	throwforce = 2
+	w_class = WEIGHT_CLASS_TINY
+	throw_speed = 3
+	throw_range = 7
+
+	///What type of power cell the cartridge has by default
+	var/obj/item/stock_parts/cell/cell
+	var/cell_type = /obj/item/stock_parts/cell
+	var/dead_cell = FALSE //set to true so the cartridge is given an empty cell
+	var/can_load = TRUE //if false, cannot load or unload cells.
+	var/can_charge = TRUE //if calse, cannot recharge in a recharger
+	var/charge_sections = 4 //Number of charge sections for visualization of remaining charge.
+	var/old_ratio = 0 // stores the cell's previous charge ratio to see if it needs an updated icon
+
+	///sound when inserting cell
+	var/load_sound = 'sound/weapons/gun/general/magazine_insert_full.ogg'
+	///sound when inserting an empty magazine
+	var/load_empty_sound = "buzz-sigh.ogg"
+	///volume of loading sound
+	var/load_sound_volume = 40
+	///whether loading sound should vary
+	var/load_sound_vary = TRUE
+
+	///Sound of ejecting a magazine
+	var/eject_sound = 'sound/weapons/gun/general/magazine_remove_full.ogg'
+	///sound of ejecting an empty magazine
+	var/eject_empty_sound = 'sound/weapons/gun/general/magazine_remove_empty.ogg'
+	///volume of ejecting a magazine
+	var/eject_sound_volume = 40
+	///whether eject sound should vary
+	var/eject_sound_vary = TRUE
+
+/obj/item/cell_cartridge/Initialize()
+	. = ..()
+	if(cell_type && !cell)
+		cell = new cell_type(src)
+
+	if(!dead_cell)
+		cell.give(cell.maxcharge)

```

Comment:
Cells by default spawn with full charge, so this doesn't work. Better to remove the `dead_cell` var and change the cell_type of `/obj/item/cell_cartridge/dead` to `/obj/item/stock_parts/cell/empty`

Reply:
Alright